pad_server/src/state.h

```diff
@@ -12,10 +14,16 @@
 typedef struct {
     bool actuators[NUM_ACTUATORS];
     arm_lvl_e arm_level;
-    pthread_mutex_t lock;
+
+    pthread_rwlock_t rw_lock;
+    pthread_rwlockattr_t rw_lock_attr;
+
+    int read_count, write_count;
```

Comment:
Please have two lines of declaration. This is best practice in C.

```c
int read_count;
int write_count;
```

Reply:
Woops, those don't actually do anything, forgot to delete them. Thanks!